src/accountLinks.js

```diff
@@ -0,0 +1,69 @@
+const { createLink } = require('3id-blockchain-utils')
+
+class AccountLinks {
+  constructor (ceramic, provider) {
+    this._ceramic = ceramic
+    this.provider = provider
+  }
+
+  async create (address, did, proof = null) {
+    const doc = await this._ceramic.createDocument(null, 'account-link', {
+      owners: [this._convertToCaip10(address)]
+    })
+    if (!proof) {
+      if (!this.provider) {
+        throw new Error('Provider must be set to create an account link')
+      }
+      proof = await createLink(did, address, this.provider)
+    }
+    await doc.change(proof)
+    return doc.content
+  }
+
+  async read (address) {
+    const docId = await this._getDocId(address)
+    const doc = await this._ceramic.loadDocument(docId)
+    await new Promise(resolve => setTimeout(resolve, 500))
```

Comment:
Why do you have a timeout here? To see if you get an update from the network?
You can use `doc.on('update', fn)` which will be triggered when there is an update that happened to the document.

Reply:
In this case, what we want is to ensure we have the most recent heads from all peers for this document. I don't think either a timeout or waiting for a single update solves the problem - this was just temporary (I should have added a comment to that effect). Any thoughts on how we want to handle this?

Reply:
I think `doc.on('update', fn)` should have the desired effect. If not we should figure out why and how to solve it better.

Reply:
Turns out that I should have been setting the `onlyGenesis` property to `false` in `loadDocument`, because it defaults to `true` - what's the use case that's designed for?

Reply:
It doesn't seem right that you would want to set `onlyGenesis` to `false`. This should only be considered when you initially create the document. When we load the document we want onlyGenesis to be true since we don't want to create an anchor record or publish the "head" (our genesis CID for the doc). 

The desired behaviour of loadDocument (without options) should be that it waits for any updates from the network and returns once it get's that update.

I guess however since you are now creating the document before with the `_getDocId` function you might be sidestepping the default behaviour.

Reply:
`update` wasn't working because the event is named 'change' :D The 'change' event does work if the document has had updates applied.

If the document *doesn't* exist, though, (the address hasn't been linked yet), if we wait for a `change` event, it waits forever (or until we update that document) - I think that's why I had used a timeout. We could combine the two (i.e. in a Promises race) but the only advantage that will have is potentially reducing the delay, while still being time-bound, and also suffering from the downside that it will assume a document is loaded on the first `change` event it receives.